go/vt/vtgate/planner.go

```diff
@@ -102,6 +102,7 @@ func (plr *Planner) WatchVSchema(ctx context.Context) {
 		plr.mu.Lock()
 		plr.vschema = vschema
 		plr.mu.Unlock()
+		plr.plans.Clear()
```

Comment:
Actually, I think it's more complicated. Let's go with this for now. I think we need a more elaborate scheme to prevent race conditions here.